server/ws/websockets.go

```diff
@@ -258,10 +263,10 @@ func (ws *Server) addListener(wsSession *websocketSession, command *WebsocketCom
 	for _, blockID := range command.BlockIDs {
 		itemID := makeItemID(workspaceID, blockID)
 		if ws.listeners[itemID] == nil {
-			ws.listeners[itemID] = []*websocket.Conn{}
+			ws.listeners[itemID] = []*Listener{}
 		}
 
-		ws.listeners[itemID] = append(ws.listeners[itemID], wsSession.client)
+		ws.listeners[itemID] = append(ws.listeners[itemID], &Listener{&sync.RWMutex{}, wsSession.client})
```

Comment:
There is a problem with this approach, you are sharing the `wsSessoin.client` and adding it to different listeners, that each one has its own Mutex, but the client is shared, so... you have still the problem of multiple writes from different places.

Probably you need to make the `websocketSession.client` a `Listener` and reuse that. (Probably the name of the type should be converted to something like `wsClient` instead of `Listener`).

Reply:
Done.